python/tvm/relay/ty.py

```diff
@@ -106,6 +122,51 @@ def __init__(self, var, kind=Kind.Type):
         self.__init_handle_by_constructor__(_make.TypeVar, var, kind)
 
 
+@register_relay_node
+class GlobalTypeVar(Type):
```

Comment:
I wasn't sure about the name either but think global type var is possibly a good choice because we can use it for other things down the line potentially.